File: BaekjoonOnlineJudge/10845/10845.cpp
#include <iostream>
#include <deque>
#include <string>

using namespace std;

int main(void)
{
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    deque<int> q;

    int N;
    cin >> N;

    while (N--)
    {
        string order;
        cin >> order;

        if (order == "push")
        {
            int num;
            cin >> num;

            q.push_back(num);
        }
        else if (order == "empty")
            cout << q.empty() << "\n";
        else if (order == "size")
            cout << q.size() << "\n";
        else if (order == "pop")
        {
            if (q.empty())
                cout << -1 << "\n";
            else
            {
                cout << q.front() << "\n";
                q.pop_front();
            }
        }
        else if (order == "front")
        {
            if (q.empty())
                cout << -1 << "\n";
            else
                cout << q.front() << "\n";
        }
        else
        {
            if (q.empty())
                cout << -1 << "\n";
            else
                cout << q.back() << "\n";
        }
    }

    return 0;
}

File: BaekjoonOnlineJudge/10845/10845.py
from collections import deque
import sys
input = sys.stdin.readline

dq = deque()
N = int(input())

for _ in range(N):
    order = input().split()

    # push
    if order[0] == "push":
        dq.append(int(order[1]))
    # pop
    elif order[0] == "pop":
        if dq:
            print(dq.popleft())
        else:
            print(-1)
    # size
    elif order[0] == "size":
        print(len(dq))
    # empty
    elif order[0] == "empty":
        if dq:
            print(0)
        else:
            print(1)
    # front
    elif order[0] == "front":
        if dq:
            print(dq[0])
        else:
            print(-1)
    # back
    else:
        if dq:
            print(dq[-1])
        else:
            print(-1)
